src/main/java/com/softserve/edu/greencity/ui/pages/econews/EcoNewsPage.java

```diff
@@ -45,13 +43,20 @@
 
     public EcoNewsPage(WebDriver driver) {
         super(driver);
-        //checkElements();
+        checkElements();
     }
 
     private void checkElements() {
+        WebElement firstItem = driver.findElement(DISPLAYED_ARTICLES.getPath());
+        try {
+            waitsSwitcher.setExplicitWait(2, ExpectedConditions.stalenessOf(firstItem));
+            logger.warn("The site performed the same GET request twice and redrew page");
+        } catch (TimeoutException error) {
+            ; //Everything is OK
+        }
```

Comment:
That was the main problem. checkElements() should be called after each EcoNewsPage update (refresh, swithing view mode etc.)